### PR TITLE
Remove incorrect task filtering

### LINKED ISSUE #39: Exclusion list is incorrect for new Android variants
### Troubleshooting

- [x] I've searched [discuss.bitrise.io](https://discuss.bitrise.io/) for possible solutions.
- Which version of the step is effected? 1.0.2
- Is the issue reproducible with the latest version? YES
- Does the issue happen sporadically, or every time? EVERY TIME
- Is the issue reproducible locally by following our [local debug guide](https://devcenter.bitrise.io/troubleshooting/debugging-your-build-on-your-own-machine/)? YES / NO / NOT APPLICABLE

### Useful information

- [FAQ](https://devcenter.bitrise.io/faq/faq-index/)
- [devcenter.bitrise.io](https://devcenter.bitrise.io/)

### Issue description

It looks like some of the Android gradle variants have been renamed (perhaps with Android Gradle Plugin 7.2.0) and therefore the exclusion list at line 90 in https://github.com/bitrise-steplib/bitrise-step-android-build/blob/master/step/step.go is insufficient.

For example there is no longer a task with the suffix `AndroidTestClasses` it is now `AndroidTestClassesToCompileJar`. I think simply adding `ClassesToCompileJar` to the exclusion list would fix it.

### Bitrise info

- Build URL: (https://app.bitrise.io/build/d6dca4f6-9a41-4159-8a21-9af4d9ac62b8)
- [Bitrise Support](https://devcenter.bitrise.io/troubleshooting/enabling-bitrise-support-user/) enabled: YES
- Log: 

<details>
  <pre>
Could not determine the dependencies of task ':app:bundleDebugUnitTestClassesToCompileJar'.
  </pre>
</details>

### Steps to reproduce

1. Create a project using Android Chipmunk and Android Gradle Plugin 7.2.0
2. Try to generate an Android aab build

### PR DESCRIPTION
**Pull request by ofalvai**
### Checklist

- [X] I've read and followed the [Contribution Guidelines](https://github.com/bitrise-steplib/.github/blob/main/CONTRIBUTING.md)
- [x] `step.yml` and `README.md` is updated with the changes (if needed)

### Version

Requires a *PATCH* [version update](https://semver.org/)

### Context

Fixes #39 by removing the variant and task filtering logic completely. The step is trying to be too clever here, we can easily build the final Gradle task name from the `module`, `variant` and `app_type` inputs.

I created a truth table of the combinations:

| module | variant (flavor+type) | task (assuming APK)|
|-- | -- | --|
|- | - | assemble|
|app | debug | :app:assembleDebug|
|app | - | :app:assemble|
|- | debug | assembleDebug|

This works correctly because of the dependency between Gradle tasks, such as:

1. Not defining a variant in the task name runs the task for all variants (such as assemble -> assembleDebug, assembleRelease)
2. Not defining a module in the task name (although not recommended) runs the task in all submodules.

The only difference between the old and this implementation: when the variant input is not defined, the old implementation selected the `:module:assembleVariantAndroidTest` and `:module:assembleVariantUnitTest` variants for execution too. I find it unnecessary and counter-intuitive, I think it's an improvement to prevent these tasks from running.

### Changes

- Remove old code that parses the available Gradle tasks and filters them according to modules and variants
- Create a simple function that builds the final Gradle tasks based on the module, variant and app_type inputs

### Decisions

Rewriting the Gradle logic in `go-android` is out of scope for now, we can gradually review the other Android steps, and once every step is using this simpler Gradle handling, we can consolidate the common implementation in the library.

### CODE REVIEW

File: e2e/bitrise.yml [hunk 1]
```diff
@@ -8,11 +8,8 @@ workflows:
     envs:
     - TEST_APP_URL: https://github.com/bitrise-io/android-multiple-test-results-sample.git
     - BRANCH: maintenance
-    - EXPECTED_APK: mylibrary-debug-androidTest.apk
```
Comment (ofalvai) [inline e2e/bitrise.yml:11]: This E2E was accidentally passing because of a bug in task creation. The step incorrectly executed additional UI testing and unit testing tasks too, which created this UI test APK in the *app module's build folder* (and not in the nested module's build folder. This is why the step was able to find and export the APK, but this was incorrect.

Turns out the artifact searching logic in this step only handles 1 level of directory depth, so it can't find the nested module's build folder (even without wildcards). I created a follow-up ticket to improve this (STEP-20355), but in the meantime I think the best workaround is to not treat export errors as step failures, allow the step to succeed in this case.

**Review by barry-irvine — APPROVED** [review]
Thank you so much for fixing this!

File: step/step_test.go [hunk 2]
```diff
@@ -262,3 +75,80 @@ func createStep() AndroidBuild {
...
+		appType string
+		module  string
+		variant string
+	}
+	tests := []struct {
+		name string
+		args args
+		want string
+	}{
+		{
```
Comment (Bence1001) [inline step/step_test.go:90]: These tests are great! One edge case I spotted was when `appType` is invalid, which is checked in this function too. I think we should have a test case for that too.

**Review by Bence1001 — APPROVED** [review]
LGTM! I have one minor comment.
